src/write/unit.rs

```diff
@@ -2809,6 +2818,18 @@ mod tests {
             add_child(unit, child1, constants::DW_TAG_variable, "grandchild1");
             add_child(unit, root, constants::DW_TAG_subprogram, "child2");
             add_child(unit, root, constants::DW_TAG_subprogram, "child3");
+
+            // Some of the following children will be deleted before writing the sections
+            let child4 = add_child(unit, root, constants::DW_TAG_subprogram, "child4");
+
+            // This will not be deleted to check that deleting interleaved children works
+            add_child(unit, root, constants::DW_TAG_subprogram, "child5");
+
+            let child6 = add_child(unit, root, constants::DW_TAG_subprogram, "child6");
+            add_child(unit, child6, constants::DW_TAG_variable, "grandchild6");
+            unit.get_mut(root).delete_child(child4);
+            // Deleting `child6` deletes `grandchild6` as well
+            unit.get_mut(root).delete_child(child6);
```

Comment:
Please don't add it in this test. This test is specifically for checking the sibling attribute, and adding this makes it too hard to follow what's going on.

I'd even be fine without a test for this given the simplicity of the code it is testing.

Reply:
That's fair, I can add a new test then. I'm hesitant to add this without a test since my previous implementation had a subtle bug when interleaving adds and deletes.